app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditDialog.java

```diff
@@ -0,0 +1,85 @@
+package com.invisibleteam.goinvisible.mvvm.edition;
+
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.app.Fragment;
+import android.content.Context;
+import android.os.Bundle;
+import android.os.Parcelable;
+
+import com.invisibleteam.goinvisible.model.InputType;
+import com.invisibleteam.goinvisible.model.Tag;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+public class EditDialog extends DialogFragment {
+
+    public static final String FRAGMENT_TAG = EditDialog.class.getName();
+    public static final String EXTRA_TAG = "extra_tag";
+
+    public static EditDialog newInstance(Context context, Tag tag) {
+        Bundle bundle = new Bundle();
+        bundle.putParcelable(EditDialog.EXTRA_TAG, tag);
+
+        return (EditDialog) Fragment.instantiate(context, FRAGMENT_TAG, bundle);
+    }
+
+    private Tag tag;
+
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        if (extractTag()) {
+            prepareView();
+        } //TODO log exception to crashlitycs on else.
+
+        return super.onCreateDialog(savedInstanceState);
+    }
+
+    private boolean extractTag() {
+        if (getArguments() != null && !getArguments().isEmpty()) {
```

Comment:
When there are more then one condition I prefer to check them with variables like: 
Object object = getArguments();
 if ( object!= null && !object.isEmpty()) {
             Parcelable extra = object.getParcelable(EXTRA_TAG);

But its not a blocker (just my fantasy ;))